Index: fuse-umfuse-ext2/fuse-ext2/Makefile.am
===================================================================
--- fuse-umfuse-ext2.orig/fuse-ext2/Makefile.am
+++ fuse-umfuse-ext2/fuse-ext2/Makefile.am
@@ -288,7 +288,7 @@ install-data-hook:
 if LINUX
 install-exec-local:
 	$(INSTALL) -d "$(DESTDIR)/usr/sbin"
-	$(LN_S) -f "$(DESTDIR)/$(bindir)/fuse-ext2" "$(DESTDIR)/sbin/mount.fuse-ext2"
+	$(LN_S) -f "$(DESTDIR)/$(bindir)/fuse-ext2" "$(DESTDIR)/$(sbindir)/mount.fuse-ext2"
 endif
 
 dist_man1_MANS = fuse-ext2.1
Index: fuse-umfuse-ext2/fuse-ext2/Makefile.in
===================================================================
--- fuse-umfuse-ext2.orig/fuse-ext2/Makefile.in
+++ fuse-umfuse-ext2/fuse-ext2/Makefile.in
@@ -1940,7 +1940,7 @@ install-data-hook:
 #endif
 @LINUX_TRUE@install-exec-local:
 @LINUX_TRUE@	$(INSTALL) -d "$(DESTDIR)/usr/sbin"
-@LINUX_TRUE@	$(LN_S) -f "$(DESTDIR)/$(bindir)/fuse-ext2" "$(DESTDIR)/sbin/mount.fuse-ext2"
+@LINUX_TRUE@	$(LN_S) -f "$(DESTDIR)/$(bindir)/fuse-ext2" "$(DESTDIR)/$(sbindir)/mount.fuse-ext2"
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
